nmigen_soc/wishbone/bus.py

```diff
@@ -166,6 +167,147 @@ def memory_map(self, memory_map):
+        self.domain = domain
+
+    def elaborate(self, platform):
+        intr_bus = self.intr_bus
+        sub_bus = self.sub_bus
+
+        m = Module()
+
+        common_addr_width = min(intr_bus.addr_width, sub_bus.addr_width)
+        m.d.comb += [
+            sub_bus.cyc.eq(intr_bus.cyc),
+            sub_bus.we.eq(intr_bus.we),
+            sub_bus.adr[(sub_bus.addr_width-common_addr_width):sub_bus.addr_width].eq(
+                intr_bus.adr[(intr_bus.addr_width-common_addr_width):intr_bus.addr_width]
+            ),
+            intr_bus.ack.eq(sub_bus.ack),
+        ]
+        if hasattr(intr_bus, "err"):
+            m.d.comb += intr_bus.err.eq(getattr(sub_bus, "err", 0))
+        if hasattr(intr_bus, "rty"):
+            m.d.comb += intr_bus.rty.eq(getattr(sub_bus, "rty", 0))
+        if hasattr(sub_bus, "lock"):
+            m.d.comb += sub_bus.lock.eq(getattr(intr_bus, "lock", 0))
+        if hasattr(sub_bus, "cti"):
+            m.d.comb += sub_bus.cti.eq(getattr(intr_bus, "cti", CycleType.CLASSIC))
+        if hasattr(sub_bus, "bte"):
+            m.d.comb += sub_bus.bte.eq(getattr(intr_bus, "bte", BurstTypeExt.LINEAR))
+
+
+        # Following code assumes single cycle read and write;
+        # duplicate check that was also done in __init__()
+        assert intr_bus.data_width <= sub_bus.data_width, "Internal error: should not happed"
+
+        # stb and stall for different pipeline combinations of initiator and subordinate bus
+        if hasattr(intr_bus, "stall") == hasattr(sub_bus, "stall"):
+            m.d.comb += sub_bus.stb.eq(intr_bus.stb)
+            if hasattr(intr_bus, "stall"):
+                m.d.comb += intr_bus.stall.eq(sub_bus.stall)
+        elif hasattr(intr_bus, "stall"):
+            # See Wishbone B4 spec: 5.2 Pipelined master connected to standard slave
+            m.d.comb += [
+                sub_bus.stb.eq(intr_bus.stb),
+                intr_bus.stall.eq(intr_bus.cyc & ~sub_bus.ack),
+            ]
+        else:
+            # See Wishbone B4 spec: 5.1 Standard master connected to pipelined slave
+            wait4ack = Signal()
+            m.d[self.domain] += wait4ack.eq(intr_bus.stb & ~(wait4ack & sub_bus.ack))
+            m.d.comb += sub_bus.stb.eq(intr_bus.stb & ~wait4ack)
+
+        # Data and sel width conversion
+        if intr_bus.addr_width == sub_bus.addr_width:
+            decoded_addr_ext = Signal()
+            m.d.comb += decoded_addr_ext.eq(1)
+        else:
+            assert intr_bus.addr_width > sub_bus.addr_width, "Internal error, should not happen"
+            m.submodules.sel_decoder = sel_decoder = SignalDecoder(width=1<<(intr_bus.addr_width-sub_bus.addr_width))
+            m.d.comb += sel_decoder.i.eq(intr_bus.adr[:intr_bus.addr_width-sub_bus.addr_width])
+            decoded_addr_ext = sel_decoder.o
+        # dat_w
+        m.d.comb += sub_bus.dat_w.eq(
+            Cat(intr_bus.dat_w for _ in range(sub_bus.data_width//intr_bus.data_width))
```

Comment:
This can be replaced by `sub_bus.dat_w.eq(Repl(intr_bus.dat_w, ...))`.

Reply:
Done. In the end I also joined different statements in one m.d.comb +=